arch/xtensa/soc/esp32/soc.h

```diff
@@ -11,6 +11,28 @@
 
 #include <zephyr/types.h>
 #include <stdbool.h>
+#include <arch/xtensa/arch.h>
+
+/**
+ * @brief Struct to peripheral masks to enable peripheral
+ * clock and reset peripherals.
+ */
+struct esp32_peripheral {
+	/** Mask for clock peripheral */
+	int clk;
+	/** Mask for reset peripheral */
+	int rst;
+};
+
+static inline void esp32_set_mask32(u32_t v, u32_t mem_addr)
```

Comment:
This could be `sys_set_mask32()` and available for everyone; what do you think?

Reply:
I'm not sure about this as it is only used by ESP32. And @tbursztyka  said:
>>> sys_io.h is meant to hold generic functions for all arch (each needing to implement these then).
>>>
>>> So either you generalize sys_set/clear_mask32 in include/sys_io.h or you just create xtensia dedicated ones (as it used to be in fact) but put them in xtensia soc.h```

If I change it to sys_set_mas32() to make available to everyone, will I need to add this function to all sys_io.h SOCs ?